### DISCUSSION TITLE
CONFIG: Oil.nvim styled floating centered Nvimtree

### DISCUSSION THREAD

**Discussion by shricodev** [nvim-tree/nvim-tree.lua#2692] [Ideas]
If you want the Oil.nvim styled nvim-tree.

> Maybe you want to use the tree-styled file floating view and with git support then this customization is super cool.

![Nvimtree config](https://github.com/nvim-tree/nvim-tree.lua/assets/76906722/8d81d051-5115-4c22-a862-21c9c3725c13)

Add the following lines to your existing configuration

```lua
        float = {
          enable = true,
          quit_on_focus_loss = true,
          -- Make the neovim window appear in the center of the screen
          open_win_config = function()
            local screen_w = vim.opt.columns:get()
            local screen_h = vim.opt.lines:get() - vim.opt.cmdheight:get()
            local window_w = screen_w * WIDTH_RATIO
            local window_h = screen_h * HEIGHT_RATIO
            local window_w_int = math.floor(window_w)
            local window_h_int = math.floor(window_h)
            local center_x = (screen_w - window_w) / 2
            local center_y = ((vim.opt.lines:get() - window_h) / 2) - vim.opt.cmdheight:get()
            return {
              border = "rounded",
              relative = "editor",
              row = center_y,
              col = center_x,
              width = window_w_int,
              height = window_h_int,
            }
          end,
        },
```

For the final config, take a look [here](https://github.com/shricodev/dotfiles/blob/dfd4143101230383ca4f2067a917b6033b92f1c4/nvim/.config/nvim/lua/shricodev/plugins/nvimtree.lua).

**Comment by alex-courtis** [comment]
This is just fantastic, the dotfiles link is most useful.

Please create a [recipe](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes) so that others may find and use this.

**Reply by shricodev** [reply]
Sure, I will. Thanks @alex-courtis 🙌
